File: src/main/java/gargoyle/rpycg/fx/FXApplication.java
package gargoyle.rpycg.fx;

import javafx.scene.Parent;

@FunctionalInterface
@SuppressWarnings({"RedundantThrows", "unused", "ProhibitedExceptionDeclared"})
public interface FXApplication {
    static void notifySplash(final FXSplash.FXSplashNotification splashNotification) {
        FXLauncher.notifySplash(splashNotification);
    }

    default void doInit() throws Exception {
    }

    FXComponent<?, ? extends Parent> doStart() throws Exception;

    default void doStop() throws Exception {
    }
}

File: fx/src/main/kotlin/gargoyle/fx/FxApplication.kt
package gargoyle.fx

import gargoyle.fx.FxSplash.FxSplashNotification
import javafx.scene.Parent

@Suppress("unused")
fun interface FxApplication {

    fun doInit() {}

    fun doStart(): FxComponent<*, out Parent>

    fun doStop() {}

    companion object {
        fun notifySplash(splashNotification: FxSplashNotification) {
            FxLauncher.notifySplash(splashNotification)
        }
    }
}
